backend/graphql/resolvers/classResolvers.ts

```diff
@@ -39,8 +40,22 @@ const classResolvers = {
     classesByTeacher: async (
       _req: undefined,
       { teacherId }: { teacherId: string },
-    ): Promise<Array<ClassResponseDTO>> => {
-      return classService.getClassesByTeacherId(teacherId);
+    ): Promise<Array<ClassCard>> => {
+      const classesByTeacher = classService.getClassesByTeacherId(teacherId);
+
+      const classCards: ClassCard[] = [];
+
+      (await classesByTeacher).forEach((classObj) => {
+        classCards.push({
+          id: classObj.id,
+          activeAssessments: classObj.testSessions.length,
+          assessmentCount: classObj.testSessions.length,
+          gradeLevel: classObj.gradeLevel,
+          className: classObj.className,
+          studentCount: classObj.students.length,
+        });
+      });
+      return classCards;
```

Comment:
```suggestion
      const classesByTeacher = await classService.getClassesByTeacherId(teacherId);
      return classesByTeacher.map((classObj) => ({
        ...classObj,
        activeAssessments: classObj.testSessions.length,
        assessmentCount: classObj.testSessions.length,
        studentCount: classObj.students.length,
      }));
```

Reply:
Sounds good! I think I will maybe merge all the props for `ClassCard` into `ClassResponseDTO` in a later ticket if we find that the query `classesByTeacher` in the resolvers is needed elsewhere and requires a `ClassResponseDTO`, or if the fields for `ClassCard` are needed in other queries. Thank you for the suggestions